fdb-record-layer-core/src/main/proto/record_metadata.proto

```diff
@@ -195,9 +195,11 @@ message JoinedRecordType {
 
   message Join {
     optional string left = 1;
-    optional KeyExpression left_expression = 2;
-    optional string right = 3;
-    optional KeyExpression right_expression = 4;
+    optional KeyExpression left_source_expression = 2;
+    optional KeyExpression left_value_expression = 3;
+    optional string right = 4;
+    optional KeyExpression right_source_expression = 5;
+    optional KeyExpression right_value_expression = 6;
```

Comment:
Are we confident that there aren't any serialized anywhere? Wouldn't it be better to have the old fields align?

Reply:
It probably would be good to align the fields/field numbers, I think, and to consider what to do if the new fields are missing to handle backwards compatibility

Reply:
done